app/forms/incomeSources/add/AddIncomeSourceStartDateForm.scala

```diff
@@ -0,0 +1,91 @@
+
+  def apply(messagesPrefix: String)(implicit messages: Messages, dateService: DateServiceInterface, dateFormatter: ImplicitDateFormatter): Form[DateFormElement] = {
+
+    val dateMustNotBeTooFarInFuture = s"$messagesPrefix.error.future"
+    val dateMustBeEntered = s"$messagesPrefix.error.required"
+    val dateMustBeReal = s"$messagesPrefix.error.invalid"
+    val dayRequired = "dateForm.error.day.required"
+    val monthRequired = "dateForm.error.month.required"
+    val yearRequired = "dateForm.error.year.required"
+    val dayAndMonthRequired = "dateForm.error.dayAndMonth.required"
+    val dayAndYearRequired = "dateForm.error.dayAndYear.required"
+    val monthAndYearRequired = "dateForm.error.monthAndYear.required"
+
+    val dayInputFieldName: String = "day"
+    val monthInputFieldName: String = "month"
+    val yearInputFieldName: String = "year"
+
+    val maximumAllowableDate: LocalDate = dateService.getCurrentDate().plusWeeks(1)
+    val maximumAllowableDatePlusOneDay: LocalDate = maximumAllowableDate.plusDays(1)
+    val futureErrorMessage: String = dateFormatter.longDate(maximumAllowableDatePlusOneDay).toLongDate
+    def dateMustNotBeInTheFuture(date: String): String = messages(dateMustNotBeTooFarInFuture, date)
+
+    def checkRequiredFields: Constraint[(String, String, String)] = Constraint("constraints.requiredFields") {
+      case (day, month, year) if day.trim.isEmpty && month.trim.isEmpty && year.trim.isEmpty =>
+        Invalid(Seq(ValidationError(dateMustBeEntered)))
+      case (day, month, year) if day.trim.isEmpty && month.trim.nonEmpty && year.trim.isEmpty =>
+        Invalid(Seq(ValidationError(dayAndYearRequired)))
+      case (day, month, year) if day.trim.isEmpty && month.trim.isEmpty && year.trim.nonEmpty =>
+        Invalid(Seq(ValidationError(dayAndMonthRequired)))
+      case (day, month, year) if day.trim.nonEmpty && month.trim.isEmpty && year.trim.isEmpty =>
+        Invalid(Seq(ValidationError(monthAndYearRequired)))
+      case (day, month, year) if day.trim.isEmpty && month.trim.nonEmpty && year.trim.nonEmpty =>
+        Invalid(Seq(ValidationError(dayRequired)))
+      case (day, month, year) if day.trim.nonEmpty && month.trim.isEmpty && year.trim.nonEmpty =>
+        Invalid(Seq(ValidationError(monthRequired)))
+      case (day, month, year) if day.trim.nonEmpty && month.trim.nonEmpty && year.trim.isEmpty =>
+        Invalid(Seq(ValidationError(yearRequired)))
+      case _ =>
+        Valid
+    }
+
+    Form(
+      mapping("income-source-start-date" -> tuple(
+        dayInputFieldName -> default(text(), ""),
+        monthInputFieldName -> default(text(), ""),
+        yearInputFieldName -> default(text(), ""))
+        .verifying(firstError(
+          checkRequiredFields,
+          validDate(dateMustBeReal)
+        )).transform[LocalDate](
+          {
+            case (day, month, year) => LocalDate.of(year.toInt, month.toInt, day.toInt)
```

Comment:
do we have validation that: year / month and day is convertible to Int and respective date exists ?